meson.build

```diff
@@ -976,7 +976,10 @@ wayland = {
              dependency('wayland-cursor', version: '>= 1.15.0', required: get_option('wayland')),
              dependency('wayland-protocols', version: '>= 1.15', required: get_option('wayland')),
              dependency('xkbcommon', version: '>= 0.3.0', required: get_option('wayland'))],
-    'header': cc.has_header('linux/input-event-codes.h', required: get_option('wayland')),
+    'header': cc.has_header('linux/input-event-codes.h', required: get_option('wayland'),
+                            # Provided by evdev-proto package on DragonFly and FreeBSD
+                            dependencies: dependency(['evdev-proto', 'wayland-client'],
+                                                     required: get_option('wayland'))),
```

Comment:
Sorry, this confused me. `evdev-proto` doesn't actually do anything here because it still has no `.pc`. I think I would prefer just linking `wayland-client` again here with a comment that it's for BSDs and leave it at that. Although it would be better for `evdev-proto` to actually ship with a .pc file so we could just do `dependencies: dependency('evdev-proto', required: false)`.

Reply:
There's already such a comment. Trying to document `evdev-proto` doesn't support pkg-config yet and how meson.build should be changed when it grows such support requires too many words than just using current version.

Reply:
`evdev-proto` here is the preferred method. I'm relying on https://github.com/mesonbuild/meson/commit/4deeb907b613

Reply:
`required: false` isn't good as it'd make BSDs try to detect `evdev-proto` even with `-Dwayland=disabled`.

Reply:
Seeing a `dependencies: dependency('evdev-proto')` in the build but knowing that it will never actually work and actually always just fallsback to `wayland-client` is confusing imo. I guess if you want to futureproof it in case `evdev-proto` magically gets support, that's OK but it wasn't obvious to me that's the intent at all.

>required: false isn't good as it'd make BSDs try to detect evdev-proto even with -Dwayland=disabled

Indeed that happens unexpectedly. Is this possibly a meson bug? What's the use of trying to find a dependency if the entire check is already disabled? It should be skipped imo. Oh well not relevant for now I guess.